src/ApplicationInsights.ServiceFabric.Native/Net45/RequestTrackingModule/ServiceRemotingServerEventListener.cs

```diff
@@ -0,0 +1,236 @@
+                {
+                    string telemetrySource = string.Empty;
+                    string sourceAppId = null;
+
+                    try
+                    {
+                        sourceAppId = ServiceRemotingHeaderUtilities.GetRequestContextKeyValue(messageHeaders, RequestResponseHeaders.RequestContextCorrelationSourceKey);
+                    }
+                    catch (Exception ex)
+                    {
+                        ServiceFabricSDKEventSource.Log.GetCrossComponentCorrelationHeaderFailed(ex.ToInvariantString());
+                    }
+
+                    string currentComponentAppId = string.Empty;
+                    bool foundMyAppId = false;
+                    if (!string.IsNullOrEmpty(rt.Context.InstrumentationKey))
+                    {
+                        foundMyAppId = this.correlationIdLookupHelper.TryGetXComponentCorrelationId(rt.Context.InstrumentationKey, out currentComponentAppId);
+                    }
+
+                    // If the source header is present on the incoming request,
+                    // and it is an external component (not the same ikey as the one used by the current component),
+                    // then populate the source field.
+                    if (!string.IsNullOrEmpty(sourceAppId)
+                        && foundMyAppId
+                        && sourceAppId != currentComponentAppId)
+                    {
+                        telemetrySource = sourceAppId;
+                    }
+
+                    string sourceRoleName = null;
+
+                    try
+                    {
+                        sourceRoleName = ServiceRemotingHeaderUtilities.GetRequestContextKeyValue(messageHeaders, RequestResponseHeaders.RequestContextSourceRoleNameKey);
+                    }
+                    catch (Exception ex)
+                    {
+                        ServiceFabricSDKEventSource.Log.GetComponentRoleNameHeaderFailed(ex.ToInvariantString());
+                    }
+
+                    if (!string.IsNullOrEmpty(sourceRoleName))
+                    {
+                        if (string.IsNullOrEmpty(telemetrySource))
+                        {
+                            telemetrySource = "roleName:" + sourceRoleName;
+                        }
+                        else
+                        {
+                            telemetrySource += " | roleName:" + sourceRoleName;
+                        }
+                    }
+
+                    rt.Source = telemetrySource;
+                }
+
+                // Call StartOperation, this will create a new activity with the current activity being the parent.
+                // Since service remoting doesn't really have an URL like HTTP URL, we will do our best approximate that for
+                // the Name, Type, Data, and Target properties
+                var operation = this.client.StartOperation<RequestTelemetry>(rt);
```

Comment:
imagine you receive request with Correlation-Context (baggage). Correlation-Context must be propagated with outgoing requests. The way outgoing request knows about baggage is through the Activity. `StartOperation` does not do that because it does not know which telemetry.Properties are set from the baggage.

So you also need to set baggage on the Activity.Current after calling `StartOperation`

Reply:
Ah - I see, so that's a bug in current V1 implementation as well. Thanks for spotting that. I'll fix it in both places.